### PR TITLE
fix: sec group from server can be None also

### PR DESCRIPTION
**Pull request by pacospace**
Introducing a fix for an edge cases + tests to avoid this in the future.

### CODE REVIEW

**Review by slothkong — COMMENTED** [review]
Requestd clarifications.

File: ewccli/backends/openstack/backend_ostack.py [hunk 1]
```diff
@@ -562,7 +563,7 @@ def list_servers(
             image_id = server.image.get("id") if server.image else None
             image_name = image_map.get(image_id, "N/A") if image_id else "N/A"
 
-            addresses = server.get("addresses")
+            addresses = getattr(server, "addresses", {}) or {}
```
[the remark this answers is not in the document]
Comment (pacospace) [reply ewccli/backends/openstack/backend_ostack.py]: should be None or {}, I will fix it.

File: ewccli/backends/openstack/backend_ostack.py [hunk 2]
```diff
@@ -580,22 +581,23 @@ def list_servers(
...
                 networks = "\n".join([n for n in server.addresses])
 
-            sec_groups = getattr(server, "security_groups", [])
+            sec_groups = getattr(server, "security_groups", None) or []
 
             servers[server.id] = {
                 "name": server.name,
                 "status": server.status,
-                "flavor": server.flavor["original_name"],
+                "flavor": server.flavor.get("original_name"),
```
[the remark this answers is not in the document]
Comment (pacospace) [reply ewccli/backends/openstack/backend_ostack.py]: I can add a warning on top and avoid the error here in case, sounds good?

File: ewccli/backends/openstack/backend_ostack.py [hunk 3]
```diff
@@ -580,22 +581,23 @@ def list_servers(
...
                     network_ip[f"{address}"] = [v.get("addr") for v in address_v]
 
-            if federee:
+            if federee and network_ip:
                 networks = "\n".join([f"{n} ({v})" for n, v in network_ip.items()])
             else:
                 networks = "\n".join([n for n in server.addresses])
 
-            sec_groups = getattr(server, "security_groups", [])
+            sec_groups = getattr(server, "security_groups", None) or []
```
[the remark this answers is not in the document]
Comment (pacospace) [reply ewccli/backends/openstack/backend_ostack.py]: sec_groups expect list in the logic, including an empty list, but definitely not a None. It's doable where to have this check but I want to keep the expected type the same even in case of None as it needs to iterate over a list after

[the remark this answers is not in the document]
Comment (slothkong) [reply ewccli/backends/openstack/backend_ostack.py]: I understand the need for the empty list default.
To reframe the question: 
Why do `getattr("something", None) or []`? Is it  not possible  to  do `getattr("something", [])` instead?

[the remark this answers is not in the document]
Comment (pacospace) [reply ewccli/backends/openstack/backend_ostack.py]: because that still return None: getattr only uses the default if the attribute does not exist, not if it exists and is None as in this case: https://docs.python.org/3/library/functions.html#getattr

**Review by slothkong — APPROVED** [review]
